easybuild/framework/easyblock.py

```diff
@@ -2171,6 +2172,14 @@ def make_module_step(self, fake=False):
                 for line in txt.split('\n'):
                     self.dry_run_msg(' ' * 4 + line)
         else:
+            module_only = build_option('module_only')
+            if module_only and os.path.exists(mod_filepath):
+               warning_msg = "Old module file found. Backing it up in %s. Diff is:\n%s"
+               mod_bck_filepath = back_up_file(mod_filepath, backup_extension="bck")
```

Comment:
Making it a hidden module makes sense just for TCL modules. I had lua modules in mind, but since there are still people out there with TCL modules I agree with you.

Regarding making it configurable..... I am not so sure.

Pros:
1. Flexibility.

Cons:
1. Yet another option cluttering the `--help`
2. More code to maintain

However, if you are using TCL modules with Lmod, these will be recognized as modules that can be loaded, which is not what you want. Any other combination (TCL modules with Tmod, or lua modules with Lmod), will result in an invisible (to the modules tool) backup. So yes, I guess I should also prepare an option for it.....